src/CMakeLists.txt

```diff
@@ -274,7 +277,12 @@ add_executable(
     ${LIBUTIL_FREEBSD_SRC}/pidfile.c)
 
 add_dependencies(tarantool build_bundled_libs)
-target_link_libraries(tarantool box ${common_libraries})
+if (TARGET_OS_DARWIN)
+    set_target_properties(tarantool PROPERTIES LINK_FLAGS "-u _box_on_shutdown")
+    target_link_libraries(tarantool box ${common_libraries} shutdown)
+else()
+    target_link_libraries(tarantool box ${common_libraries} -Wl,--whole-archive shutdown -Wl,--no-whole-archive)
```

Comment:
Could you please explain why we have to use --whole-archive and -Wl...

Reply:
box_on_shutdown function not used in tarantool code. In case of LTO build linker removes this function from tarantool executable. There are two ways to prevent it: use -u or --whole-archive linker options. For mac there is no --whole-archive options, so we used -u. For other systems we used --whole-archive because -u option needs function name as in object file not in source! I see that in different systems we have different function name in object file, so i think it's better to use --whole-archive, than to do "elseif" for each system in CMakeLists

Reply:
Also there is no really unused functions in this library!

Reply:
See exports.h. You don't need to patch any build options.

Reply:
Thanks! Fixed!